src/figuresClasses.ts

```diff
@@ -1,11 +1,108 @@
-export interface Figure {}
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+enum ShapeEnum {
+  TRIANGLE = 'triangle',
+  CIRCLE = 'circle',
+  RECTANGLE = 'rectangle',
+}
+
+export interface Figure {
+  shape: ShapeEnum;
+  color: ColorType;
+  getArea: () => number;
+}
+
+export abstract class BaseFigure implements Figure {
+  shape: ShapeEnum;
+
+  color: ColorType;
+
+  constructor(shape: ShapeEnum, color: ColorType) {
+    this.color = color;
+    this.shape = shape;
+  }
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier, TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property. You don't need to manually assign them as in lines 21 and 22.